src/components/portrait/portrait.pw.tsx

```diff
@@ -0,0 +1,486 @@
+
+      await expect(tooltip).toBeVisible();
+      await expect(tooltip).toHaveId(tooltipId);
+    });
+  });
+
+  visibilityValues.forEach(([boolVal, renderState]) => {
+    test(`should render ${renderState} a tooltip, when visibility prop is ${boolVal}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <PortraitComponent tooltipMessage="foo" tooltipIsVisible={boolVal} />
+      );
+
+      const tooltip = getDataElementByValue(page, "tooltip");
+
+      if (boolVal) {
+        await expect(tooltip).toBeVisible();
+        await expect(tooltip).toHaveText("foo");
+      } else {
+        await expect(tooltip).not.toBeVisible();
+      }
+    });
+  });
+
+  ["top", "bottom", "left", "right"].forEach((tooltipPosition) => {
+    test(`should render with tooltip positioned ${tooltipPosition}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <Box p={200}>
+          <PortraitComponent tooltipPosition={tooltipPosition} />{" "}
+        </Box>
+      );
+
+      const tooltip = getDataElementByValue(page, "tooltip");
+
+      await expect(tooltip).toBeVisible();
+      await expect(tooltip).toHaveAttribute("data-placement", tooltipPosition);
+    });
+  });
+
+  test("should render with a tooltip error", async ({ mount, page }) => {
+    await mount(<PortraitComponent tooltipType="error" />);
+
+    const tooltip = getDataElementByValue(page, "tooltip");
+
+    await expect(tooltip).toHaveCSS("background-color", VALIDATION.ERROR);
+  });
+
+  [
+    [SIZE.MEDIUM, "14px"],
+    [SIZE.LARGE, "16px"],
+  ].forEach(([tooltipSize, fontSize]) => {
+    test(`should render with a ${tooltipSize} size tooltip`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<PortraitComponent tooltipSize={tooltipSize} />);
+
+      const tooltip = getDataElementByValue(page, "tooltip");
+
+      await expect(tooltip).toBeVisible();
+      await expect(tooltip).toHaveCSS("font-size", fontSize);
+    });
+  });
+
+  colors.forEach(([names, color]) => {
+    test(`should render with a tooltip with a ${names} background color`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<PortraitComponent tooltipBgColor={color} />);
+
+      const tooltip = getDataElementByValue(page, "tooltip");
+
+      await expect(tooltip).toBeVisible();
+      await expect(tooltip).toHaveCSS("background-color", color);
+    });
+  });
+
+  colors.forEach(([names, color]) => {
+    test(`should render with a tooltip with a ${names} font color`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<PortraitComponent tooltipFontColor={color} />);
+
+      const tooltip = getDataElementByValue(page, "tooltip");
+
+      await expect(tooltip).toBeVisible();
+      await expect(tooltip).toHaveCSS("color", color);
+    });
+  });
+
+  test("should render and call onClick callback when a click event", async ({
+    mount,
+    page,
+  }) => {
+    let callBackCount = 0;
```

Comment:
```suggestion
    let callbackCount = 0;
```

**nitpick** (non-blocking) - "callback" is a single word (even though I guess it's originally made up of 2), maybe it's just me but camel-casing it as `callBack` looks really weird to me